unic/utils/src/traits.rs

```diff
@@ -0,0 +1,21 @@
+//! # Unic - Utils - Traits
+//!
+//! A component of [`unic`: Unicode and Internationalization Crates for Rust](/unic/).
+//!
+//! Defines traits used extensively across UNIC.
+
+use std::{fmt, str};
+
+/// A Property that applies to a character
+pub trait CharProperty : str::FromStr + fmt::Display {
```

Comment:
What I'm thinking that we should do is to get this part in, have the trait implemented manually, and switch to the macro in a separate step.

Also, we need to make sure we differentiate between numerical properties, complex enum properties, and flat enum properties. For example, `fn all_values()` should only be defined for flat enum properties.

I'll also submit my ideas for classifying properties with traits soon.